libsql/internal/http/http.go

```diff
@@ -3,61 +3,90 @@ package http
 import (
 	"bytes"
 	"context"
+	"database/sql/driver"
 	"encoding/json"
 	"fmt"
 	"io"
 	"net/http"
+	"sort"
 	"time"
 
 	"github.com/xwb1989/sqlparser"
 )
 
 var httpClient = &http.Client{Timeout: 120 * time.Second}
 
-type params struct {
-	Names  []string
-	Values []any
-}
+type paramsType int
 
-func (p *params) MarshalJSON() ([]byte, error) {
-	if len(p.Values) == 0 {
-		return json.Marshal([]string{})
-	}
-	if len(p.Names) == 0 {
-		return json.Marshal(p.Values)
+const (
+	namedParameters paramsType = iota
+	positionalParameters
+)
+
+type parameters struct {
+	paramsValues []struct {
+		name  string
+		value any
 	}
-	m := map[string]interface{}{}
-	for idx := range p.Values {
-		m["@"+p.Names[idx]] = p.Values[idx]
+	paramsType paramsType
+}
+
+func (p *parameters) MarshalJSON() ([]byte, error) {
```

Comment:
There's no point in repacking the params again. Can't we structure them in parameters in such a way that we can just call marshal on them? for example:
```
type params struct {
  positional []any
  named [string]any
}
```
with the assumption that always only one of them is not empty